### PR TITLE
Migrate pylint check to Python 3

### PR DESCRIPTION
**Pull request by hcho3**
As we are about to deprecate Python 2 (#4379), we need to migrate pylint (Python coding style check) to Python 3. There are some minor changes to meet the new requirements.

### CODE REVIEW

File: python-package/xgboost/core.py [hunk 1]
```diff
@@ -1721,7 +1714,7 @@ def get_split_value_histogram(self, feature, fmap='', bins=None, as_pandas=True)
         xgdump = self.get_dump(fmap=fmap)
         values = []
         regexp = re.compile(r"\[{0}<([\d.Ee+-]+)\]".format(feature))
-        for i in range(len(xgdump)):
+        for i, _ in enumerate(xgdump):
```
Comment (trivialfis) [inline python-package/xgboost/core.py]: Just out of curiosity, why using `enumerate`.

File: python-package/xgboost/training.py [hunk 2]
```diff
@@ -462,7 +462,7 @@ def cv(params, dtrain, num_boost_round=10, nfold=3, stratified=False, folds=None
                                rank=0,
                                evaluation_result_list=res))
         except EarlyStopException as e:
-            for k in results.keys():
+            for k in results:
```
Comment (trivialfis) [inline python-package/xgboost/training.py]: Does this change anything related to code logic?

Comment (hcho3) [reply python-package/xgboost/training.py]: No, the change is to comply with the latest pylint guideline (PyCQA/pylint#699). Logic-wise nothing changed.

File: python-package/xgboost/core.py (hunk 1 above)
Comment (hcho3) [reply python-package/xgboost/core.py]: To comply with pylint style guideline: PyCQA/pylint#684

**Comment by hcho3** [comment]
@trivialfis I think this is ready. I'd like to get this merged soon, since #4380 depends on it.

**Pull request merged**